app.py

```
@@ -102,357 +125,270 @@ def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
+        go.Figure: Results table for the D-Wave hybrid solver.
+        str: Class name for the D-Wave tab.
+        str: The label for the D-Wave tab.
+        bool: True if D-Wave tab should be disabled, False otherwise.
+        bool: Whether D-Wave solver is running.
     """
-    if run_click == 0 or ctx.triggered_id != "run-button":
-        empty_figure = get_empty_figure("Run optimization to see results")
-        return empty_figure, "tab"
-    if ctx.triggered_id == "run-button":
-        if "Hybrid" in solver:
-            model_data = JobShopData()
-            filename = SCENARIOS[scenario]
-            model_data.load_from_file(DATA_PATH.joinpath(filename), resource_names=RESOURCE_NAMES)
-            allow_quadratic_constraints = model == "QM"
-            start = time.time()
-            results = run_shop_scheduler(
-                model_data,
-                use_mip_solver=False,
-                allow_quadratic_constraints=allow_quadratic_constraints,
-                solver_time_limit=time_limit,
-            )
-            end = time.time()
-            fig = generate_gantt_chart(df=results, y_axis="Job", color="Resource")
-            table = generate_output_table(results["Finish"].max(), time_limit, int(end - start))
-            return fig, table, "tab-success", {"visibility": "visible"}
-        else:
-            time.sleep(0.1)
-            empty_figure = get_empty_figure(
-                HTML_CONFIGS["solver_messages"]["dwave"]["solver_not_chosen"]
-            )
-            table = generate_output_table(0, 0, 0)
-            return empty_figure, table, "tab-warning", {"visibility": "hidden"}
-    else:
+    if ctx.triggered_id != "run-button" or run_click == 0:
         raise PreventUpdate
 
+    if not SamplerType.HYBRID.value in solvers:
+        return (dash.no_update, dash.no_update, "tab", "D-Wave Results", True, False)
+
+    if isinstance(model, int):
+        model = Model(model)
```

What happens if `model="a"` is passed in? You might want to raise `ValueError` for invalid inputs.

---

This function is only used by Dash (it's never called by us or any user anywhere) so `model` should only ever be an int. It's probably safe enough not doing an instance check at all, but I agree that that a more robust type check would otherwise be preferred here.

---

Updated to remove check: https://github.com/dwave-examples/job-shop-scheduling-cqm/pull/17/commits/5e41bd2e6de63f470bc2ac073b985861d81f7698